src/main/scala/sigmastate/eval/CompiletimeCosting.scala

```diff
@@ -112,9 +141,34 @@ trait CompiletimeCosting extends RuntimeCosting { IR: Evaluation =>
         val index = fn.substring(1).toByte
         eval(mkSelectField(tuple.asTuple, index))
 
-      case Terms.Apply(Select(col, "slice", _), Seq(from, until)) =>
+      case Select(obj, method, Some(tRes: SNumericType))
+        if obj.tpe.isNumType && obj.asNumValue.tpe.isCastMethod(method) =>
```

Comment:
@greenhat Please use Array in `isCastMethod` and also avoid allocation on every method call.

Reply:
Fixed.